homeassistant/components/neato/vacuum.py

```diff
@@ -35,15 +41,59 @@
 ATTR_CLEAN_SUSP_COUNT = 'clean_suspension_count'
 ATTR_CLEAN_SUSP_TIME = 'clean_suspension_time'
 
+ATTR_MODE = 'mode'
+ATTR_NAVIGATION = 'navigation'
+ATTR_CATEGORY = 'category'
+ATTR_ZONE = 'zone'
+
+SERVICE_NEATO_CUSTOM_CLEANING = 'neato_custom_cleaning'
+
+SERVICE_NEATO_CUSTOM_CLEANING_SCHEMA = VACUUM_SERVICE_SCHEMA.extend({
```

Comment:
I don't think we should extend this vacuum schema. It has entity_id as optional and allows the `all` keyword, which we don't support here. We could support it, but it would mean more code here.

I suggest we put entity_id as required and validate with `cv.entity_ids`.

Reply:
Done